src/odemis/gui/cont/acquisition.py

```diff
@@ -2141,3 +2147,47 @@ def _on_ac_done(self, future):
         self._tab_panel.lbl_auto_center.Show()
         self._tab_panel.gauge_auto_center.Hide()
         self._sizer.Layout()
+
+
+class FastEMAlignmentController:
+    def __init__(self, tab_data, panel):
+        self._tab_data = tab_data
+        self._panel = panel
+
+        self._panel.align_gauge_progress.Hide()
+        panel.btn_align.Bind(wx.EVT_BUTTON, self._on_btn_align)
+
+        self._is_aligning = False
+
+    def _on_btn_align(self, evt):
+        if self._is_aligning:
+            logging.error("Cancelling alignment currently not supported.")
+            return
+        self._is_aligning = True
+
+        # Change button to "cancel", show progress bar
+        self._panel.align_gauge_progress.Show()
+        # Label is displayed on the right, but during the acquisition, we want the gauge bar to occupy the
+        # full available space --> use a panel instead of a spacer which can be hidden (spacers cannot) to
+        # make it look nice.
+        self._panel.align_spacer_panel.Hide()
+        self._panel.align_lbl_gauge.Hide()
+        self._panel.btn_align.SetLabel("Cancel")
+        self._panel.Parent.Layout()
```

Comment:
Are you sure that `self._panel.Layout()` is not sufficient? I don't get why you have you force the layout of the parent (but I know that wxWidgets is sometimes strange about this).